api/src/main/java/com/netflix/iceberg/expressions/UnboundPredicate.java

```diff
@@ -67,11 +68,13 @@ Expression bind(Types.StructType struct) {
    */
   public Expression bind(Types.StructType struct, boolean caseSensitive) {
     Types.NestedField field;
-    if (caseSensitive) {
-      field = struct.field(ref().name());
-    } else {
-      field = struct.caseInsensitiveField(ref().name());
-    }
+    String expressionFieldPath = ref().name();
+
+    boolean isStructFieldExp = expressionFieldPath.indexOf('.') > -1 &&
+      struct.field(expressionFieldPath.split("\\.")[0]).type() instanceof Types.StructType;
```

Comment:
Nit: in terms of where we want to put in the constraint of filtering only against struct types, I'm wondering if it'd make sense to only code this aspect in the private `findStructField` method.  

We'd only have `bind` delegate to `findStructField` (findNestedField ?)  in the case that the `expressionFieldPath` contains nested fields only so we'd simplify the condition as `boolean isFieldExp = expressionFieldPath.indexOf('.') > -1`.

I'm thinking that if we may end up adding support for other nested types (i.e. map) we'd make changes only to the `findStructField` implementation without having to change the internals of `bind` to delegate to a potential private method `findMapField` and adding cyclomatic complexity to evaluating the `field` attribute.

Wdyt?

Reply:
I like the idea. adding it in.

Reply:
done.